### PR TITLE
fix(dialect/field): convert arith.select carrying binary field types

### PR DESCRIPTION
**Pull request by chokobole**
## What

`BinaryFieldToArith` marks the whole arith dialect statically legal, so the
type-polymorphic `arith.select` is never rewritten even when its operands
carry binary field types (`arith.select i1, !field.bf<3>, !field.bf<3>`).
The operands *do* get converted, and the framework's compensating
`unrealized_conversion_cast` (storage int ↔ `!field.bf`) survives to the end
of the conversion:

```
error: failed to legalize unresolved materialization from ('i8') to ('!field.bf<3>')
       that remained live after conversion
```

This is exactly the carve-out `ModArithToArith` already has for the same
reason — mirror it:

```cpp
target.addDynamicallyLegalOp<arith::SelectOp>(
    [&](auto op) { return typeConverter.isLegal(op); });
```

Prime fields never hit this because `FieldToModArith` doesn't blanket-legalize
arith and `ModArithToArith` has the carve-out; every binary field (t3/t7/ghash)
failed.

## Why

fractalyze/xla#205 — XLA:CPU fused select+reduce over binary fields
(the `jnp.where(mask, x, 0)` + `jnp.sum` shape emitted by HLO fusions) fails
to compile with the error above. This pass-legality fix is the root cause;
the XLA side only adds regression tests (plus a pin bump once this lands via
stablehlo).

## Test

- New lit cases `@test_bf_select` (tower, i8) and `@test_bf_ghash_select`
  (flat GHASH basis, i128) in `tests/Dialect/Field/binary_field_to_arith.mlir`;
  both fail with the unresolved-materialization error before the fix.
- Full suite: 111/111 pass.
- Downstream (with this checkout overriding the pin): the three new XLA CPU
  select+reduce fusion tests for t3/t7/ghash flip FAIL → PASS, and the GPU
  mirrors pass.

### CODE REVIEW

File: prime_ir/Dialect/Field/Conversions/BinaryFieldToArith/BinaryFieldToArith.cpp [hunk 1]
```diff
@@ -523,6 +523,12 @@ struct BinaryFieldToArith : impl::BinaryFieldToArithBase<BinaryFieldToArith> {
     target.markUnknownOpDynamicallyLegal(
         [&](Operation *op) { return typeConverter.isLegal(op); });
 
+    // arith::SelectOp is in the legal ArithDialect but may carry binary
+    // field types (e.g., arith.select i1, !field.bf<3>, !field.bf<3>).
+    // Override the dialect-level legal status so it gets converted.
+    target.addDynamicallyLegalOp<arith::SelectOp>(
+        [&](auto op) { return typeConverter.isLegal(op); });
```
[the remark this answers is not in the document]
Comment (chokobole) [reply prime_ir/Dialect/Field/Conversions/BinaryFieldToArith/BinaryFieldToArith.cpp]: Superseded in the updated revision: rather than carving `arith.select` out of the blanket-legal arith dialect, the fix now deletes `addLegalDialect<arith::ArithDialect>()` itself — the pass's existing `markUnknownOpDynamicallyLegal` + identity-fallback type converter already give every arith op the right dynamic legality, matching `FieldToModArith` (the pass that runs immediately before this one). The line this comment targets no longer exists.

For the record, the suggestion as written does not compile: with an explicit `arith::SelectOp` parameter, `typeConverter.isLegal(op)` is ambiguous — `SelectOp` implicitly converts to both `Operation *` and `Value`, and `TypeConverter::isLegal` has overloads reachable from both. That is also why the identical carve-out in `ModArithToArith.cpp` uses `auto` (it deduces `Operation *` through the legality-callback wrapper). The `ConstantOp`/`AddOp`/`UnrealizedConversionCastOp` callbacks in this file can use explicit types because they call `containsBinaryFieldType`, not `isLegal`.

**Review by batzor — APPROVED** [review]
